plugins/flow-visibility/policy-recommendation/policy_recommendation_job.py

```diff
@@ -0,0 +1,577 @@
+import antrea_crd
+import datetime
+import json
+import kubernetes.client
+import random
+import string
+import uuid
+import yaml
+from ipaddress import ip_address, IPv4Address
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import udf
+from pyspark.sql.types import StringType
+from re import sub
+
+# Column names of flow record table in Clickhouse database used in recommendation job 
+FLOW_TABLE_COLUMNS = [
+    'sourcePodNamespace',
+    'sourcePodLabels',
+    'destinationIP',
+    'destinationPodNamespace',
+    'destinationPodLabels',
+    'destinationServicePortName',
+    'destinationTransportPort',
+    'protocolIdentifier',
+]
+
+NAMESPACE_ALLOW_LIST = [
+    'kube-system',
+    'flow-aggregator',
+    'flow-visibility'
+]
+
+def get_flow_type(destinationServicePortName, destinationPodLabels):
+    if destinationServicePortName != "":
+        return "pod_to_svc"
+    elif destinationPodLabels != "":
+        return "pod_to_pod"
+    else:
+        return "pod_to_external"
+
+def map_flow_to_egress(flow):
+    src = "#".join([flow.sourcePodNamespace, flow.sourcePodLabels])
```

Comment:
Is `#` a self-defined delimiter? If so, I think we'd better define it as a constant.